src/Omega_h_simModel2d.cpp

```diff
@@ -0,0 +1,466 @@
+#include <SimModel.h>
+#include <SimUtil.h>
+#include "Omega_h_model2d.hpp"
+#include "Omega_h_profile.hpp"
+#include "Omega_h_map.hpp"  // get_degrees
+#include "Omega_h_adj.hpp"  // invert_adj
+#include "Omega_h_array_ops.hpp" // operator==(LOs,LOs)
+#include <map>
+#include <algorithm> //std::fill
+#include <numeric> //std::exclusive_scan
+
+namespace Omega_h {
+
+HostWrite<LO> createAndInitArray(size_t size, const LO init=0) {
+  auto array = HostWrite<LO>(size);
+  std::fill(array.data(), array.data()+array.size(), init);
+  return array;
+}
+
+HostWrite<LO> createArray(size_t size) {
+  return HostWrite<LO>(size);
+}
+
+bool isModel2D(pGModel mdl) {
+  return (GM_numRegions(mdl) == 0);
+}
+
+bool isValid(pGModel mdl, bool checkGeo = false) {
+  const auto maxCheckLevel = 2; //may be expensive
+  auto geoCheck = checkGeo ? maxCheckLevel : 0;
+  pPList errList =  NULL;
+  const auto valid = 1;
+  return (GM_isValid(mdl,geoCheck,errList) == valid);
+}
+
+struct EntInfo {
+  LOs ids;
+  std::map<int,int> idToIdx;
+};
+
+struct VtxInfo : public EntInfo {
+  using EntType = pGVertex;
+  Reals coords;
+};
+
+struct EdgeInfo : public EntInfo {
+  using EntType = pGEdge;
+};
+struct FaceInfo : public EntInfo {
+  using EntType = pGFace;
+};
+
```

Comment:
placeholder to remind myself: 

I've yet to look at the rest of this file beyond this line, I'll get back to this soon as I can